### PR TITLE
Update for Ilias Tübingen

### PR DESCRIPTION
**Pull request by Tyxz**
I wrote a little update, because the login form in Tübingen has changed apparently.
Additionally, I fixed a little "bug". Someone uploaded a file which ended with '.' in one of my courses. This lead to a sync error and, not sure why, also lead to an error loop while the downloader tried to sync the file again and again.
My fix is to simply check all files if they end with a '.' and just strip the dot in the local copy of it.

### CODE REVIEW

File: src/main/java/plugin/TuebIlias.java [hunk 1]
```diff
@@ -40,17 +41,17 @@ public LoginStatus login(String username, String password) {
 
 			Document doc = Jsoup.parse(html);
 			Element form = doc.select("form[action*=idp").first();
-			Element input_tag = form.select("input[name='lt']").first();
-			String ltKey = input_tag.attr("value");
-			
+
+			// Updated 06.02.2020 to fit the new login process
```
Comment (DeOldSax) [inline src/main/java/plugin/TuebIlias.java]: no need to put a comment with dates for updates - git is your friend :)

File: src/main/java/download/IliasFileDownloaderTask.java [hunk 2]
```diff
@@ -36,6 +36,10 @@ protected Void call() throws Exception {
 			entity = response.getEntity();
 
 			BufferedInputStream in = new BufferedInputStream(entity.getContent());
+			// remove invalid file endings
+			String targetPath = this.targetPath.endsWith(".")
+					? this.targetPath.substring(0, this.targetPath.length() - 1)
+					: this.targetPath;
```
Comment (DeOldSax) [inline src/main/java/download/IliasFileDownloaderTask.java]: Good catch! What do you think about changing this in [IliasScraper](https://github.com/DeOldSax/iliasDownloaderTool/blob/2b5af55ff504ab71fe38f6a6e039aeb5deeaba23/src/main/java/control/IliasScraper.java#L177) directly when the file extension is scraped?

File: src/main/java/plugin/TuebIlias.java [hunk 3]
```diff
@@ -40,17 +41,17 @@ public LoginStatus login(String username, String password) {
...
-			
-			nvps.add(new BasicNameValuePair("lt", ltKey));
+			nvps.add(new BasicNameValuePair("j_username", username));
+			nvps.add(new BasicNameValuePair("j_password", password));
+
 			nvps.add(new BasicNameValuePair("execution", "e1s1"));
-			nvps.add(new BasicNameValuePair("_eventId", "submit"));
+			nvps.add(new BasicNameValuePair("donotcache", "1"));
+			nvps.add(new BasicNameValuePair("_eventId_proceed", ""));
 			post.setEntity(new UrlEncodedFormEntity(nvps, Consts.UTF_8));
```
Comment (DeOldSax) [inline src/main/java/plugin/TuebIlias.java]: i guess i will blindly trust you with that changes :) thx!

File: README.md [hunk 4]
```diff
@@ -1,29 +1,53 @@
...
+   With `sdk list java` or  `sdk list maven`, you can find the other available versions.
+3. Create a`JAVA_HOME` variable:
+    ```shell
+   echo JAVA_HOME="$HOME/.sdkman/candidates/java/current/bin/" >> ~/.bashrc
+   source ~/.bashrc
+    ```
+4. Clone this repo and change your university in the `pom.xml` at line 12 and in `src/main/java/control/IliasManager.java` at line 14++.
+5. Build it with `mvn clean install`.
+6. Start it with `java -jar ./release/IliasDownloaderTool-ube-v1.2.2//Ilias-Downloader-Tool-ube-v1.2.2.jar`.
+
```
Comment (DeOldSax) [inline README.md]: typo: //

actually steps 4.-6. also apply on windows or macos - maybe the initial creator was too lazy to give a hint how to change the university :P Thank you for adding this!

File: src/main/java/plugin/TuebIlias.java (hunk 3 above)
Comment (Tyxz) [reply src/main/java/plugin/TuebIlias.java]: Hehe, it works, I promise. At least it does for me now ;) 
Not sure with the `nvps.add(new BasicNameValuePair("donotcache", "1"));` though. It tells the server to not set a cookie I think. Did not test it. Set it to 1 and it worked^^

File: src/main/java/download/IliasFileDownloaderTask.java (hunk 2 above)
Comment (Tyxz) [reply src/main/java/download/IliasFileDownloaderTask.java]: You are much more familiar with the code base. If you think it should be included into the Scraper, you are most likely right. I just was in need of a quick fix to be able to bulk donwload from Ilias and wanted to share a working solution. To include this the right way, is too time consuming right now. Sorry. Maybe after my exams.

File: src/main/java/plugin/TuebIlias.java (hunk 1 above)
Comment (Tyxz) [reply src/main/java/plugin/TuebIlias.java]: I looked at the GPL2 and was not sure if I had to mark the changes^^

File: README.md (hunk 4 above)
Comment (Tyxz) [reply README.md]: You're right. I copied it from my terminal. Shouldn't have a typo, but there definitely is one.
It's possible the other steps could work under Windows as well. I think there is even a Windows version of SDKMAN!. Only the installation process and setting the path variable should be changed.

**Comment by Tyxz** [comment]
Thanks for the review @DeOldSax and thank you for this wonderful software. It makes it so much easier to download a seminar folder.
I will fix the typo, try my luck at a better implementation of the dot fix and maybe add a Windows explanation after my exams.

**Comment by Tyxz** [comment]
I updated the Readme again and also added an step for Windows into the "Build with SDKMAN!" example. Additionally, I removed the unnecessary comments from Tuebilias and removed the warnings from some duplicated code and unused imports.
To remove the hack from IliasFileDownloaderTask, I guess you could add a line like this to IliasScraper: `if (fileExtension.equals("")) fileExtension = "undefined";`, but am not sure how to test it. The erroneous file was fixed and I can't upload a new one myself. I just have to assume, that the fileExtension is empty if a file extension is missing in the response. Therefore, I would let it be like this, until someone with write access to Ilias could try out this or another solution.

**Comment by DeOldSax** [comment]
will be merged in the next release! thx again!

**Comment by schtandard** [comment]
@DeOldSax When will the next release be? Logging in to the Tübingen Ilias does not work currently (and I suspect this would fix it)..

**Comment by DeOldSax** [comment]
@schtandard  thx for mentioning - i will have an eye on it!

**Comment by DeOldSax** [comment]
merged with 486f315

**Pull request closed without merging**